samples/WebApi/README.md

```diff
@@ -0,0 +1,127 @@
+> cd myApp
+```
+
+## Add CoreRT to your project
+
+Using CoreRT to compile your application is done via the ILCompiler NuGet package.
+
+In your shell/command prompt navigate to the root directory of your project run the command 
+
+`> dotnet new nuget `
+
+This will add a nuget.config file to your application. Open the file and in the ``<packageSources> `` element under ``<clear/>`` add the following:
+
+```xml
+<add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" /> 
+<add key="aspnet-core" value="https://dotnet.myget.org/F/aspnetcore-release/api/v3/index.json" /> 
+```
+
+Once you've added the package source, add a reference to the compiler by running the following command:
+
+`> dotnet add package Microsoft.DotNet.ILCompiler -v 1.0.0-alpha-* `
+
+After the package has been succesfully added to your project, open the file called `Startup.cs` and in the `ConfigureServices()` method modify the line:    
+    
+```csharp
+services.AddMvc(); 
+```
+
+to
+
+```csharp 
+services.AddMvcCore().AddJsonFormatters(); 
+```
+
+## Using reflection 
+Runtime directives are XML configuration files, which specify which elements of your program are available for reflection. They are used at compile-time to enable AOT compilation in applications at runtime. 
+
+In this sample a basic rd.xml file has been added for a simple Web API application under the root project folder. Copy its contents to your application directory and modify the
+```xml <Assembly Name="SampleWebApi" Dynamic="Required All" /> ``` element to use your app's name.
+
+If your application makes use of reflection, you will need to create a rd.xml file specifying explicitly which assemblies and types should be made available. For example, in  your .NET Core Web API application, reflection is required to determine the correct namespace, from which to load the ``Startup`` type. Both are defined respectively via the `<Assembly>` and `<Type>` attributes. For example, in the case of our specific application;
+
+```xml 
+<Assembly Name="SampleWebApi">
+  <Type Name="SampleWebApi.Startup" Dynamic="Required All" />
+</Assembly> 
+```
+
+At runtime, if a method or type is not found or cannot be loaded, an exception will be thrown. The exception message will contain information on the missing type reference, which you can then add to the rd.xml of your program.
+
+Once you've created a rd.xml file, navigate to the root directory of your project and open its `.csproj` file and in the first `<PropertyGroup>` element add the following:
+
+```xml
+<RdXmlFile>path_to_rdxml_file\rd.xml</RdXmlFile>
+<RuntimeIdentifiers>runtime_identifier</RuntimeIdentifiers>
+```
+
+where path_to_rdxml_file is the location of the file on your disk and runtime_identifier is one of win-x64, linux-x64 or osx-x64, depending on the OS for which you would like to publish. Under the second `<ItemGroup>` remove the line containing a reference to `Microsoft.AspNetCore.All` and substitute it with:
+
```

Comment:
It is really needed to replace `Microsoft.AspNetCore.All` with the list of packages? If it is really needed, it would be useful to explain why.

Reply:
Not necessary, but similarly to `Mvc` vs `MvcCore`  it reduces the final closure - `Microsoft.AspNetCore.All` has references to a lot of packages we don't need, which also makes `dotnet restore` even slower than it is (and ILCompiler package is pretty chunky)
Added an explanation with a reference to the issue describing why we're only adding explicitly necessary services.